queue_processors/autoreduction_processor/post_process_admin.py

```diff
@@ -179,12 +179,42 @@ def _load_reduction_script(self, instrument_name):
         """ Returns the path of the reduction script for an instrument. """
         return os.path.join(self._reduction_script_location(instrument_name), 'reduce.py')
 
-    def send_reduction_started(self):
-        """Log and update AMQ message to reduction started"""
-        logger.debug("Calling: %s\n%s",
-                     ACTIVEMQ_SETTINGS.reduction_started,
-                     self.message.serialize(limit_reduction_script=True))
-        self.client.send(ACTIVEMQ_SETTINGS.reduction_started, self.message)
+    def send_reduction_message(self, message, amq_message):
+        """Send/Update AMQ reduction message
+        :param message: (str) amq reduction  status
+        :param amq_message: (str) reduction status path
+        """
+        try:
+            logger.debug("Calling: %s\n%s",
+                         amq_message,
+                         self.message.serialize(limit_reduction_script=True))
+            self.client.send(amq_message, self.message)
+            logger.info("Reduction: %s", message)
+
+        except AttributeError:
+            logger.debug("Failed to find send reduction message: %s", amq_message)
+
+    def determine_reduction_status(self):
+        """
+        Determine which message type to log and send to AMQ, triggering exception if job failed
+        """
+        if self.message.message is not None:
```

Comment:
could we rename `self.message.message` to something more meaningful/obvious?

Reply:
self.message is the Message class.
self.message.message refers to the message attribute within the message class for an AMQ message located here `model/message/message.py`.  Changing this to be something more meaningful/obvious should be its own issue to be handled separately. This is a good point to raise as the name is't completely obvious.